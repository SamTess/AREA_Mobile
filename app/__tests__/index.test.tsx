import { render } from '@testing-library/react-native';
import React from 'react';
import Index from '../index';

jest.mock('expo-router', () => {
  const actual = jest.requireActual('expo-router');
  const { Text } = require('react-native');
  return { ...actual, Redirect: ({ href }: any) => <Text>{`redirect:${href}`}</Text> };
});

describe('Index route', () => {
  it('redirects to tabs', () => {
    const { getByText } = render(<Index />);
    expect(getByText('redirect:/(tabs)')).toBeTruthy();
  });

  it('renders without errors', () => {
    const { toJSON } = render(<Index />);
    expect(toJSON()).toBeTruthy();
  });

  it('performs redirect correctly', () => {
    const { getByText } = render(<Index />);
    const redirectText = getByText('redirect:/(tabs)');
    expect(redirectText).toBeDefined();
    expect(redirectText.props.children).toBe('redirect:/(tabs)');
  });

  it('renders Redirect component', () => {
    const { toJSON } = render(<Index />);
    expect(toJSON()).toMatchSnapshot();
  });
});
