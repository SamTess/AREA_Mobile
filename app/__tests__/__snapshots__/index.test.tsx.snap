// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index route renders Redirect component 1`] = `
<Text>
  redirect:/(tabs)
</Text>
`;
